src/celeritas/optical/OpticalDistributionData.hh

```diff
@@ -45,5 +48,22 @@ struct OpticalDistributionData
     }
 };
 
+//---------------------------------------------------------------------------//
+/*!
+ * Data used by \c OpticalStepCollector class implementation.
+ */
+struct OpticalStepCollectorData
+{
+    real_type time{};  //!< Pre-step time
+    real_type step_length{};  //!< Step length
```

Comment:
You got me thinking. The `OpticalStepCollectorData` only existed because we'd need a `StepInterface` callback class. I guess if we have `SimTrackView`, we'd also have `GeoTrackView` to provide the position, since we'll probably have access to `CoreState` data. Should I make this independent from a callback class and assume we have global core params/state directly?

Reply:
Well we'll still need the pre-step data which we'll have to cache somewhere, so I think having this for the time/step points makes sense.

Reply:
Correct. But since we are now down to just the points, I could just label it as a `Result = EnumArray<StepPoint, OpticalStepData>` of the class, which removes an indirection and uses an existing data struct created for the `OpticalDistributionData`.

Reply:
Yes, good idea.